packages/graphql-auth-transformer/src/AuthRule.ts

```diff
@@ -2,7 +2,12 @@ export type AuthStrategy = 'owner' | 'groups' | 'public' | 'private';
 export type AuthProvider = 'apiKey' | 'iam' | 'oidc' | 'userPools';
 export type ModelQuery = 'get' | 'list';
 export type ModelMutation = 'create' | 'update' | 'delete';
-export type ModelOperation = 'create' | 'update' | 'delete' | 'read';
+export enum ModelOperation {
+  CREATE = 'create',
```

Comment:
I disagree with converting the union type to enum. It is either way just a compile time safety, at runtime you'll operate with strings only, especially that we're mapping the string values to their GraphQL directive counterparts. This change breaks the pattern we're using in all our transformers. On the other hand enum values should be Pascal cased.

Reply:
string enums do allow you to access the value at runtime (as I do below using ModelOperation.CREATE). Of course this transpiles to a raw string, but IMO it makes the code more readable because you don't have to use raw strings for comparison and makes it obvious that the value is an enum and not just a string.

I do agree it breaks the existing pattern and consistency is good, but I think for the above reasons as we make changes over time we should migrate to the string enum pattern rather than the string union pattern

Reply:
I'm all in to do a refactor and use enums, but in a separate PR targeting refactoring, partial changes are confusing IMHO ;-)